source/sync/migrate-sync-modes.txt

```diff
@@ -0,0 +1,271 @@
+.. _realm-sync-migrate-modes:
+
+=========================
+Migrate Device Sync Modes
+=========================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 3
+   :class: singlecol
+
+Atlas App Services Apps can have one of two Device Sync modes: Partition-Based Sync
+and Flexible Sync. Partition-Based Sync is an older mode and you should consider
+migrating to Flexible Sync.
+
+Migrating an App that uses Partition-Based Sync to Flexible Sync is an automatic
+process. Other than upgrading the SDK version, migrating Sync modes doesn't require 
+any changes to your client app code.
+
+Migration results in Partition-Based Sync clients communicating with a
+Flexible Sync backend. We recommend eventually migrating your client apps to also
+use Flexible Sync instead of Partition-Based Sync.
+
+Requirements
+------------
+
+App Services App:
+
+- Currently use Partition-Based Sync
+
+Linked Atlas Cluster:
+
+- Use MongoDB 5.0 or later for Flexible Sync compatibility
+- Have storage auto-scaling enabled
+- Be an M10+ Cluster Tier
+- Have time-based oplog setting enabled
+
+Client apps:
+
+- Use ``realm-core`` vTBD+, which means these SDK versions: TBD
+
+.. _realm-sync-migrate-what-to-know:
+
+What to Know Before Migrating
+-----------------------------
+
+Migrating your App's Sync mode involves changes that affect both client apps and
+your App's backend. You should be aware of the side effects and plan ahead.
+
+- How long a migration takes is directly proportional to the amount of data that
+  needs to be migrated. The more data, the longer it will take to migrate.
+- Migrations can be :ref:`canceled <realm-sync-migrate-cancel>` during the 
+  migration or :ref:`reverted <realm-sync-migrate-revert>` within a certain 
+  timeframe after the migration is finished.
+- Progress notifications will not work after migration because Flexible Sync does
+  not support them.
+- Client apps that use an SDK with with a realm-core version lower than TBD will
+  experience a client reset after migration. You should use a 
+  :ref:`client reset <handle-a-client-reset>` handler with recovery enabled.
+- :ref:`Development Mode <development-mode>` is not supported during a migration.
```

Comment:
This is a bit misleading - dev mode can still be enabled on the backend, and client apps can still add new columns to existing tables without issues. The scenario that @michael-wb and @danieltabacaru were saying would not work properly is if a client tried to create a new table against an FLX backend while using a PBS config locally.

I don't think it's worth getting into all of the nuances about what exact behavior they will see, so maybe something simple like `Creating new tables via Development Mode may lead to unexpected behavior while using a partition-based sync configuration in your app`

Reply:
Ha! Yeah, my first local revision for this had a ton of detail. I decided to pare it back and went too far. Thanks.